wolfcrypt/src/tfm.c

```diff
@@ -2945,8 +2960,8 @@ int fp_sqr(fp_int *A, fp_int *B)
     y = A->used;
 
     /* call generic if we're out of range */
-    if (y + y > FP_SIZE) {
-       err = fp_sqr_comba(A, B);
+    if (y + y >= FP_SIZE) {
+       err = FP_VAL;
```

Comment:
Why is generic no longer an option?  The comment is also wrong if we're removing the generic option.

Reply:
The fp_sqr_comba is called at the end of the function.
This was checking whether the result was too big and doing it anyway.
It looks like it would have done it twice too!

Reply:
Can you fix the comment then too since we're no longer calling the generic version here?

Reply:
Fixed the comment.

Reply:
Thanks!